business/definitionService.js

```diff
@@ -104,12 +119,10 @@ class DefinitionService {
 
     const foundDefinitions = flatten(await Promise.all(concat(promises)))
     // Filter only the revisions matching the found definitions
-    return coordinatesList.filter(coordinates => {
-      return (
-        foundDefinitions.findIndex(definition => definition === EntityCoordinates.fromObject(coordinates).toString()) >
-        -1
-      )
-    })
+    return intersectionBy(
+      map(coordinatesList, coordinates => EntityCoordinates.fromObject(coordinates).toString()),
+      foundDefinitions
+    )
```

Comment:
Agreed. Lodash intersectionBy here is far more efficient and syntactically concise